File: lib-okhttp/src/main/java/com/androidx/h5/utils/OkHttpUtil.java
package com.androidx.h5.utils;

import java.util.concurrent.TimeUnit;

import code.util.LogUtil;
import code.util.MySSLSocketClient;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

public class OkHttpUtil {

    private static OkHttpClient sOkHttpClient;
    private static final int DEFAULT_READ_TIMEOUT = 8000; //读取超时，单位毫秒
    private static final int DEFAULT_CONNECT_TIMEOUT = 8000; //连接超时，单位毫秒

    public static OkHttpClient getOkHttpClient() {
        if (sOkHttpClient != null) {
            return sOkHttpClient;
        }
        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        if (LogUtil.isDebug()) { //这行必须加 不然默认不打印
            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        } else {
            interceptor.setLevel(HttpLoggingInterceptor.Level.NONE);
        }
        sOkHttpClient = new OkHttpClient.Builder()
                .addInterceptor(interceptor)
                .connectTimeout(DEFAULT_CONNECT_TIMEOUT, TimeUnit.MILLISECONDS)
                .readTimeout(DEFAULT_READ_TIMEOUT, TimeUnit.MILLISECONDS)
                .sslSocketFactory(MySSLSocketClient.getSSLSocketFactory(), MySSLSocketClient.getX509TrustManager())
                .hostnameVerifier(MySSLSocketClient.getHostnameVerifier())
                .build();
        return sOkHttpClient;
    }
}

File: app-sdk/src/main/java/book/sdk/util/OkHttpUtil.kt
package book.sdk.util

import book.sdk.core.util.TestUtil
import book.util.MySSLSocketClient.getX509TrustManager
import book.util.TlsSniSocketFactory
import book.util.TrueHostnameVerifier
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import java.util.concurrent.TimeUnit

object OkHttpUtil {
    private val mHostOkHttpClient = HashMap<String, OkHttpClient?>()
    private const val CONNECT_TIMEOUT = 8000
    private const val READ_TIMEOUT = 8000
    fun getOkHttpClient(domain: String): OkHttpClient? {
        var okHttpClient = mHostOkHttpClient[domain]
        if (okHttpClient != null) {
            return okHttpClient
        }
        val interceptor = HttpLoggingInterceptor()
        if (TestUtil.isLoggable()) { //这行必须加 不然默认不打印
            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY)
        } else {
            interceptor.setLevel(HttpLoggingInterceptor.Level.NONE)
        }
        okHttpClient = OkHttpClient.Builder()
            .addInterceptor(interceptor)
            .sslSocketFactory(TlsSniSocketFactory(domain), getX509TrustManager())
            .hostnameVerifier(TrueHostnameVerifier(domain))
            .connectTimeout(CONNECT_TIMEOUT.toLong(), TimeUnit.MILLISECONDS)
            .readTimeout(READ_TIMEOUT.toLong(), TimeUnit.MILLISECONDS)
            .retryOnConnectionFailure(false)
            .build()
        mHostOkHttpClient[domain] = okHttpClient
        return okHttpClient
    }
}
